storage/remote/queue_manager.go

```diff
@@ -2151,22 +2135,14 @@ func buildWriteRequest(logger *slog.Logger, timeSeries []prompb.TimeSeries, meta
 		return nil, highest, lowest, err
 	}
 
-	// snappy uses len() to see if it needs to allocate a new slice. Make the
-	// buffer as long as possible.
-	if buf != nil {
-		*buf = (*buf)[0:cap(*buf)]
-	} else {
-		buf = &[]byte{}
-	}
-
-	compressed, err = compressPayload(buf, pBuf.Bytes(), enc)
+	compressed, _, err = compression.Encode(compr, pBuf.Bytes(), buf)
```

Comment:
If we're using Snappy, is it possible that we return false? Am I missing something, or shouldn't we check for a potential false in the second return here?

Reply:
I am removing compress bool, it creates more problems than it helps (: